sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -603,6 +603,30 @@ static bool compatibleWithDevice(RTDeviceBinaryImage *BinImage,
   return (0 == SuitableImageID);
 }
 
+// The input BinImage should be a devicelib image for bfloat16 conversions,
+// there will be 2 bfloat16 devicelib images for fallback and native version.
+// The native version must be used on platforms with bfloat16 extension.
+static bool selectBF16Devicelib(RTDeviceBinaryImage *BinImage,
```

Comment:
This function doesn't really select anything so the name is a little bit confusing. However, at the moment of writing this comment I don't have a good suggestion yet.

Reply:
The function has been merged into isSpecialImageSHouldBeUsed.